pkg/admissioncontroller/webhooks/auth/seed/handler.go

```diff
@@ -0,0 +1,232 @@
+		return
+	}
+
+	// Decode request body into authorizationv1beta1.SubjectAccessReviewSpec structure
+	sarSpec, gvk, err := h.decodeRequestBody(body)
+	if err != nil {
+		h.logger.Error(err, "unable to decode the request")
+		h.writeResponse(w, nil, Errored(http.StatusBadRequest, err))
+		return
+	}
+
+	// Log request information
+	keysAndValues := []interface{}{"user", sarSpec.User, "groups", sarSpec.Groups}
+	if sarSpec.ResourceAttributes != nil {
+		keysAndValues = append(keysAndValues, "resourceAttributes", sarSpec.ResourceAttributes.String())
+	}
+	if sarSpec.NonResourceAttributes != nil {
+		keysAndValues = append(keysAndValues, "nonResourceAttributes", sarSpec.NonResourceAttributes.String())
+	}
+	h.logger.V(1).Info("received request", keysAndValues...)
+
+	// Consult authorizer for result and write the response
+	ctx, cancel := context.WithTimeout(context.Background(), DecisionTimeout)
+	defer cancel()
+
+	decision, reason, err := h.authorizer.Authorize(ctx, AuthorizationAttributesFrom(*sarSpec))
+	if err != nil {
+		h.logger.Error(err, "error when consulting authorizer for opinion")
+		h.writeResponse(w, gvk, Errored(http.StatusInternalServerError, err))
+		return
+	}
+
+	var status authorizationv1beta1.SubjectAccessReviewStatus
+	switch decision {
+	case authorizer.DecisionAllow:
+		status = Allowed()
+	case authorizer.DecisionDeny:
+		status = Denied(reason)
+	case authorizer.DecisionNoOpinion:
+		status = NoOpinion(reason)
+	default:
+		status = Errored(http.StatusInternalServerError, fmt.Errorf("unexpected decision: %d", decision))
+	}
+
+	h.logger.V(1).Info("responding to request", append([]interface{}{"decision", decision, "reason", reason}, keysAndValues...))
+	h.writeResponse(w, gvk, status)
+}
+
+func (h *handler) writeResponse(w io.Writer, gvk *schema.GroupVersionKind, status authorizationv1beta1.SubjectAccessReviewStatus) {
+	// The SubjectAccessReviewStatus type is exactly the same for both authorization.k8s.io/v1beta1 and
+	// authorization.k8s.io/v1, so we can just overwrite the apiVersion/kind with the same version like the object in
+	// the request.
+	response := authorizationv1beta1.SubjectAccessReview{Status: status}
+	if gvk == nil || *gvk == (schema.GroupVersionKind{}) {
+		response.SetGroupVersionKind(authorizationv1beta1.SchemeGroupVersion.WithKind("SubjectAccessReview"))
+	} else {
+		response.SetGroupVersionKind(*gvk)
+	}
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		h.logger.Error(err, "unable to encode the response")
+		h.writeResponse(w, gvk, Errored(http.StatusInternalServerError, err))
+		return
+	}
+
+	if log := h.logger; log.V(1).Enabled() {
+		log = log.WithValues(
+			"allowed", status.Allowed,
+			"denied", status.Denied,
+			"reason", status.Reason,
+			"error", status.EvaluationError,
+		)
+		log.V(1).Info("wrote response")
+	}
+}
+
+func (h *handler) decodeRequestBody(body []byte) (*authorizationv1beta1.SubjectAccessReviewSpec, *schema.GroupVersionKind, error) {
```

Comment:
We not `authorizationv1`? 👀

Reply:
According to https://kubernetes.io/docs/reference/access-authn-authz/webhook/#request-payloads:

> When faced with an authorization decision, the API Server POSTs a JSON- serialized authorization.k8s.io/v1beta1 SubjectAccessReview object describing the action.

Reply:
Okay, thanks. It's also [configureable](https://github.com/kubernetes/kubernetes/pull/84768) since `v1.17`, so we can keep it in mind when we drop `v1.16` support for Garden clusters.